src/library/scala/Array.scala

```diff
@@ -61,7 +61,35 @@ object Array extends FallbackArrayBuilding {
   val emptyShortArray   = new Array[Short](0)
   val emptyObjectArray  = new Array[Object](0)
 
-  implicit def canBuildFrom[T](implicit t: ClassTag[T]): CanBuildFrom[Array[_], T, Array[T]] =
+  implicit def canBuildFrom[T](implicit t: ClassTag[T]): CanBuildFrom[Array[_], T, Array[T]] = {
+    val tag = implicitly[ClassTag[T]]
+    (tag.runtimeClass match {
```

Comment:
Is this match really an improvement on the JVM? Because at least on JS this is going to be detrimental.

Reply:
its a saving on the allocation of the CanBuildFrom, not the CPU

Reply:
previously a use of a CBF would switch on the runtime class twice for each array, this changes it it 3 times. If its really an issue then we can specialise the code for each case and reduce to calls to one for each usage

what is the measured effect on JS?

Reply:
Well in JS de can entirely stack allocate the `CanBuildFrom` and the `ArrayBuilder` down to completely eliminating them and constant-folding the `ClassTag`s. We can do that because they are new instances. If they first switch to fetch an existing instance on the heap, we don't know what's in the heap and we can't constant-fold its members.

But I realized that we already override `Array.scala` with our own implementation to achieve this. So I guess the change in this PR won't affect us. At least not beyond the fact that it widens the gap between the JVM and the JS implementation of the library.

Reply:
Also, on the JVM, it is best to first switch on `tag.runtimeClass.isPrimitive` to have a fast path for non-primitive cases.

Reply:
@sjrd I'd be open to keeping an `private[scala] final val isScalaJS = false` constant in the library if that would help you avoid copy/paste.

Reply:
I did consider the isPrimitive check, but wanted to discuss. 
Will rework, but it should also apply in other cases. There seem to be 9 locations to apply this

Reply:
I see that @retronym fixed these